From 6f89c9abb5f3638ce8c6a875b734777ed81b654f Mon Sep 17 00:00:00 2001
From: Simeon Marijon <simeon.marijon@foss.st.com>
Date: Thu, 21 Sep 2023 09:39:01 +0200
Subject: [PATCH 0510/1695] dts: stm32: declare the reboot modes in
 stm32mp131.dtsi

reboot mode is a consumer of nvmem cell

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 7bc1e0f3193b9bd6c8bb09e6e53c92f3ee3a970e dts: stm32: declare the reboot modes in stm32mp131.dtsi]
Change-Id: Ic71f623939aa85b91f5f75bcb2adc8f78cc57038
Signed-off-by: Simeon Marijon <simeon.marijon@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/328928
Reviewed-by: Simeon MARIJON <simeon.marijon@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Simeon MARIJON <simeon.marijon@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
(cherry picked from commit e6858dd9bb997cb3a61cbea047dc9fbd417ace1f)
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/351775
---
 core/arch/arm/dts/stm32mp131.dtsi | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp131.dtsi b/core/arch/arm/dts/stm32mp131.dtsi
index f22f1962e..6e91fd80d 100644
--- a/core/arch/arm/dts/stm32mp131.dtsi
+++ b/core/arch/arm/dts/stm32mp131.dtsi
@@ -377,6 +377,20 @@
 					reg = <0x7c 0x4>;
 				};
 			};
+
+			reboot_mode: reboot-mode {
+				compatible = "nvmem-reboot-mode";
+				nvmem-cells = <&boot_mode>;
+				nvmem-cell-names = "reboot-mode";
+				mode-normal = <0x00>;
+				mode-fastboot = <0x01>;
+				mode-recovery = <0x02>;
+				mode-stm32cubeprogrammer = <0x03>;
+				mode-ums_mmc0 = <0x10>;
+				mode-ums_mmc1 = <0x11>;
+				mode-ums_mmc2 = <0x12>;
+				mode-romcode_serial = <0xff>;
+			};
 		};
 
 		pinctrl: pin-controller@50002000 {
-- 
2.47.2

